core/src/test/java/io/substrait/relation/utils/StringHolder.java

```diff
@@ -0,0 +1,37 @@
+package io.substrait.relation.utils;
+
+import com.google.protobuf.Any;
+import io.substrait.relation.Extension;
+import java.util.Objects;
+
+/**
+ * For use in {@link io.substrait.relation.ProtoRelConverterTest}
+ *
+ * <p>Used to verify serde of {@link com.google.protobuf.Any} fields in the spec
+ */
+public class StringHolder implements Extension.Enhancement, Extension.Optimization {
```

Comment:
Just curious if an extension protobuf.Any was partly composed of already supported operators (or other know objects), do you think that the processing could still take place smoothly?

Reply:
I don't think I can speak to a general case as that's going to depend on how you've structured your message. This effectively just gives you a hook where you have access to the underlying Any. It should be flexibly enough to allow you convert your data into whatever you need. Something like:
```java
  @Override
  protected Extension.Enhancement enhancementFromAdvancedExtension(Any any) {
    try {
      var unpackedMessage = any.unpack(YourMessage.class)
       <further processing into your own custom representation>
    } catch (InvalidProtocolBufferException e) {
      throw new RuntimeException(e);
    }
  }


```